tools/latte-assembler/src/gfd.cpp

```diff
@@ -325,7 +493,7 @@ gfdAddPixelShader(gfd::GFDFile &file,
 
    out.regs.sq_pgm_resources_ps = out.regs.sq_pgm_resources_ps
       .NUM_GPRS(numGpr)
-      .STACK_SIZE(1);
+      .STACK_SIZE(out.loopVars.size() * 2);
```

Comment:
I'm still not entirely sure about this one. there is definitely some correlation between loopVarsCount and stacksize though.